src/basic/string-table.h

```diff
@@ -115,3 +115,17 @@ ssize_t string_table_lookup(const char * const *table, size_t len, const char *k
                 }                                                       \
                 funlockfile(stdout);                                    \
         } while (false)
+
+#define DUMP_STRING_TABLE2(name,type,min,max)                           \
+        do {                                                            \
+                flockfile(stdout);                                      \
+                for (type _k = (min); _k < (max); _k++) {               \
+                        const char *_t;                                 \
+                        _t = name##_to_string(_k);                      \
+                        if (!_t)                                        \
+                                continue;                               \
+                        fputs_unlocked(_t, stdout);                     \
+                        fputc_unlocked('\n', stdout);                   \
+                }                                                       \
+                funlockfile(stdout);                                    \
+        } while (false)
```

Comment:
Please make `DUMP_STRING_TABLE` take a `min` arg, instead of duplicating the whole thing.
(Or maybe rename `DUMP_STRING_TABLE` to `DUMP_STRING_TABLE_FULL` with the arg and make `DUMP_STRING_TABLE` a thin wrapper.)